tests/MongoDB.Driver.Core.Tests/Core/Connections/BinaryConnection_CommandEventTests.cs

```diff
@@ -142,11 +156,26 @@ public void Should_process_a_redacted_command(string commandName)
                 responseTo: requestMessage.RequestId);
             ReceiveMessages(replyMessage);
 
+
+            var conditionallySecuritySensitiveCommands =
+                CommandEventHelperReflector.__conditionallySecuritySensitiveCommands;
+            var commandStartedCommandShouldNotBeRedacted =
+                conditionallySecuritySensitiveCommands.ContainsKey(commandName) &&
+                !conditionallySecuritySensitiveCommands[commandName](command);
+            var commandSucceededCommandShouldNotBeRedacted =
+                conditionallySecuritySensitiveCommands.ContainsKey(commandName) &&
+                !conditionallySecuritySensitiveCommands[commandName](replyMessage.Documents[0]);
+
+            var commandStartedCommandExpectedElementCount =
+                commandStartedCommandShouldNotBeRedacted ? command.ElementCount : 0;
+            var commandSucceededCommandExpectedElementCount =
+                commandSucceededCommandShouldNotBeRedacted ? reply.ElementCount : 0;
+
```

Comment:
I think lines 160-172 can be simplified to:

    var commandStartedCommandShouldBeRedacted = CommandEventHelperReflector.ShouldRedactMessage(commandName, command);
    var commandSucceededCommandShouldBeRedacted = CommandEventHelperReflector.ShouldRedactMessage(commandName, replyMessage.Documents[0]);
    var commandStartedCommandExpectedElementCount = commandStartedCommandShouldBeRedacted ? 0 : command.ElementCount;
    var commandSucceededCommandExpectedElementCount = commandSucceededCommandShouldBeRedacted ? 0 : reply.ElementCount;

Reply:
Note the change of variable names to remove "Not" from a boolean.